### ISSUE TITLE
[FEAT] Ability to specify which react platform to run cli commands on

### ISSUE THREAD

**Issue by mikeyfarina** [refinedev/refine#4795]
### Is your feature request related to a problem? Please describe.

file structure
```
...
apps
-- client (next app)
-- admin (refine app using vite)
...
package.json (shared between apps and contains both next and vite)
```

Yes, it is related to a problem. I am migrating a refine app built with Vite into an Nx monorepo that has a single shared package.json in the root. The Nx monorepo also contains a Next.js app.

It seems like when I run the `refine dev` command, the cli attempts to start the app as a next.js app rather than a vite app.

### Describe alternatives you've considered

I am thinking I will have to switch to using the nextjs platform to run refine on, rather than using vite (which I'd prefer and already have set up).

### Additional context

The problem seems to be coming from the `getProjectType` util in the cli package.

this is a snippet of code from line 26:

```
    // check for next
    if (dependencies.includes("next") || devDependencies.includes("next")) {
        return ProjectTypes.NEXTJS;
    }

    // check for remix
    if (
        dependencies.includes("@remix-run/react") ||
        devDependencies.includes("@remix-run/react")
    ) {
        return ProjectTypes.REMIX;
    }

    // check for vite
    if (dependencies.includes("vite") || devDependencies.includes("vite")) {
        return ProjectTypes.VITE;
    }
```

I believe that due to next being listed as a dependency within the large, shared package.json of the monorepo, the `getProjectType` util returns that the refine app requires the Next.js config rather than the Vite config.

### Describe the thing to improve

I would like to be able to pass in a flag that specifies the proper config to build/dev correctly.

ex: `--platform=vite`

**Comment by BatuhanW** [comment]
Hello @mikeyfarina, thanks for the detailed issue. Your suggestion looks cool! Would you like to contribute to refine with this feature? If so, we'd be more than happy to help you through the process.

**Comment by MahirMahdi** [comment]
If @mikeyfarina decides to pass on it, I would like to work on it.

**Comment by mikeyfarina** [comment]
I'd love to! Just a heads up: I haven't worked on a large OSS project much, but I'd stoked to help out!

**Comment by mikeyfarina** [comment]
@MahirMahdi I actually don't think I'll have enough time this week to implement it. Would you like to take it over?

**Comment by MahirMahdi** [comment]
> @MahirMahdi I actually don't think I'll have enough time this week to implement it. Would you like to take it over?

Sure, I'd be more than happy to take it over.

**Comment by BatuhanW** [comment]
@MahirMahdi assigned this issue to you, looking forward to your PR 🙌🏼

**Comment by MahirMahdi** [comment]
> @MahirMahdi assigned this issue to you, looking forward to your PR 🙌🏼

According to the suggestion, this is the solution I came up with: 

    /// read scripts from package.json
    const scripts = getScripts();

    // check if specified platform exists
     const platform = Object.values(scripts)
    .map(script => script.match(/--platform=([^ ]+)/)?.[1]).filter(platform => platform)

    if (platform.length > 0) {
        const projectType: keyof typeof ProjectTypes = platform[0]?.toUpperCase() as keyof typeof ProjectTypes;
        return ProjectTypes[projectType];
    }

But I'm having trouble testing it locally. I've bootstrapped and built the packages and it's showing in the terminal that the changes are being applied to the packages. However, I console logged some values to check whether everything's working fine or not, but the logged values are not showing.

Sorry something came up and I couldn't start earlier.

It would be great if you could help me test it locally so that I could make the PR without wasting any more time.

**Comment by BatuhanW** [comment]
Make sure you bootstrap a `@refinedev/cli` package and then do `npm run start -- --scope @refinedev/cli`, so your changes will be reflected.

**Comment by MahirMahdi** [comment]
I'm sorry if I wasn't clear enough earlier. What I meant was I followed all those steps and it even logged in the terminal that changes were detected. 

![Screenshot 2023-08-14 203809](https://github.com/refinedev/refine/assets/66949598/036269a4-4774-4529-b17f-07540c68733f)

What I meant to ask is how do I test or make sure that the changes I've made work properly?

**Comment by BatuhanW** [comment]
You shouldn't parse package.json file to get this parameter. You should utilize cli parameter and use it.

`refine dev --platform vite`

**Comment by mikeyfarina** [comment]
I don't mind taking a shot at this today if that works for you @MahirMahdi, I have a ton of free time!

**Comment by mikeyfarina** [comment]
@BatuhanW I'm running into the same issue that @MahirMahdi did. I built and bootstrapped all apps, and made changes that registered in the cli console. I'm having trouble understanding how to actually run the command on the bootstrapped cli. Mind giving a little insight? 

How do I run `refine dev --platform vite` on the locally ran cli?

**Comment by BatuhanW** [comment]
Once you built the cli or used start script, you can navigate to cli directory.

```
cd packages/cli

node dist/cli.js dev --platform vite
```

So basically, replace `refine` part with `node dist/cli.js`.

**Comment by mikeyfarina** [comment]
Hi, I have a fix committed successfully, but I can't seem to open a PR to submit my work, please help!

**Comment by mikeyfarina** [comment]
[Here is a PR in my forked repo](https://github.com/refinedev/refine/pull/4833/files) for the time being, any comments on the code are appreciated!

**Comment by MahirMahdi** [comment]
Sorry @BatuhanW  for not making a PR on time. From now on I'll be active. Looking forward to contributing to refine and being a part of the community.

**Comment by aliemir** [comment]
The fix is released with `@refinedev/cli@2.9.0` 🎉
